src/averageAges.js

```diff
@@ -55,7 +72,24 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon === undefined
+    ? [...people]
+    : people.filter(child => child.sex === 'm');
+
+  let ageDifference = children.map(child => getAgeDifference(child));
+
+  function getAgeDifference(child) {
+    const mother = people.find(person => person.name === child.mother);
+
+    return mother && child.born - mother.born;
+  }
+
+  ageDifference = ageDifference.filter(el => el);
```

Comment:
What's the purpose of this filtering? Do you really need it?

Reply:
The getGeDifference function returns age difference (number) or 'undefined'. So i need to get rid of 'undefined' in the array to avoid mistakes calculating averageAgeDifference.